app/models/piece.rb

```diff
@@ -1,8 +1,105 @@
-class Piece < ApplicationRecord
-  belongs_to :user
-  belongs_to :game
-
-  def self.types
-    %w(King Queen Rook Bishop Knight Pawn)
-  end
-end
+class Piece < ApplicationRecord
+  belongs_to :user
+  belongs_to :game
+
+  def self.types
+    %w[King Queen Rook Bishop Knight Pawn]
+  end
+
+  def obstructed?(destination_x, destination_y)
+    @destination_x = destination_x
+    @destination_y = destination_y
+
+    destination_tests = [horizontal?, vertically?, diagonal_up_right_down_left?, diagonal_up_left_down_right?]
+
+    if destination_tests.include?(true)
+      return true
+    elsif invalid_input?(destination_tests)
+      return 'Error: Invalid Input'
+    else
+      false
+    end
+  end
+
+  def invalid_input?(destination_tests)
+    invalid_input = @destination_y > 7 || @destination_x > 7 || @destination_y < 0 || @destination_x < 0
+    return true if (!destination_tests.include?(true) && !destination_tests.include?(false)) || invalid_input
+  end
+
+
+  def horizontal?

```

Comment:
@RaykPlatzek I don't understand what you mean.

Take a look here: https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/

Git can't merge the PR because it doesn't know how to merge what is in this branch with master.